driver-core/src/test/java/com/datastax/driver/core/ProtocolBetaVersionTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import org.testng.annotations.Test;
+
+import static com.datastax.driver.core.ProtocolVersion.V4;
+import static com.datastax.driver.core.ProtocolVersion.V5;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ * Tests for the new USE_BETA flag introduced in protocol v5
+ * and Cassandra 3.10.
+ */
+@CCMConfig(version = "3.10")
+public class ProtocolBetaVersionTest extends CCMTestsSupport {
+
+    /**
+     * Verifies that the driver can connect to 3.10 with the following combination of options:
+     * Version V5
+     * Flag SET
+     * Expected version: V5
+     *
+     * @jira_ticket JAVA-1248
+     */
+    @Test(groups = "short")
+    public void should_connect_with_beta_when_beta_version_explicitly_required_and_flag_set() throws Exception {
+        Cluster cluster = Cluster.builder()
+                .addContactPoints(getContactPoints())
+                .withPort(ccm().getBinaryPort())
+                .allowBetaProtocolVersion()
+                .build();
+        cluster.connect();
```

Comment:
Also oddly against `cassandra-3.11` this test currently fails, here's the error on the C* side:

```
ERROR [Native-Transport-Requests-2] 2017-01-05 21:54:33,272 Message.java:623 - Unexpected exception during request; channel = [id: 0xc8ec7c60, L:/127.0.1.1:60045 - R:/127.0.1.1:60141]
java.lang.IndexOutOfBoundsException: readerIndex(50) + length(4) exceeds writerIndex(51): SlicedAbstractByteBuf(ridx: 50, widx: 51, cap: 51/51, unwrapped: PooledUnsafeDirectByteBuf(ridx: 118, widx: 160, cap: 1024))
        at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1179) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:638) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.buffer.AbstractByteBuf.readUnsignedInt(AbstractByteBuf.java:646) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at org.apache.cassandra.cql3.QueryOptions$Codec.decode(QueryOptions.java:405) ~[main/:na]
        at org.apache.cassandra.cql3.QueryOptions$Codec.decode(QueryOptions.java:366) ~[main/:na]
        at org.apache.cassandra.transport.messages.QueryMessage$1.decode(QueryMessage.java:48) ~[main/:na]
        at org.apache.cassandra.transport.messages.QueryMessage$1.decode(QueryMessage.java:44) ~[main/:na]
        at org.apache.cassandra.transport.Message$ProtocolDecoder.decode(Message.java:281) ~[main/:na]
        at org.apache.cassandra.transport.Message$ProtocolDecoder.decode(Message.java:262) ~[main/:na]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:88) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:280) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:396) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:572) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:513) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:427) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:399) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_112]
```